src/escape.c

```diff
@@ -373,6 +385,49 @@ bool checkEscapeRef(Scope *sc, Expression *e, bool gag)
             e->e1->accept(this);
             e->e2->accept(this);
         }
+
+        void visit(CallExp *e)
+        {
+            /* If the function returns by ref, check each argument that is
+             * passed as 'return ref'.
+             */
+            Type *t1 = e->e1->type->toBasetype();
+            TypeFunction *tf = (TypeFunction *)t1;
```

Comment:
I would rather you checked the type before casting, instead of creating an invalid pointer then correcting it later when `t1` is a delegate.